services/task-svc/internal/task/task.go

```diff
@@ -0,0 +1,240 @@
+package task
+
+import (
+	"context"
+	"github.com/google/uuid"
+	zlog "github.com/rs/zerolog/log"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"hwgorm"
+	"task-svc/api"
+	intPatient "task-svc/internal/patient"
+)
+
+type Base struct {
+	Name        string `gorm:"column:name"`
+	Description string `gorm:"column:description"`
+	Status      uint32 `gorm:"column:status"`
+}
+
+type Task struct {
+	Base
+	Id             uuid.UUID     `gorm:"column:id"`
+	AssignedUserId uuid.NullUUID `gorm:"column:assigned_user_id;default:NULL"`
+	PatientId      uuid.UUID     `gorm:"column:patient_id"`
+}
+
+type ServiceServer struct {
+	api.UnimplementedTaskServiceServer
+}
+
+func NewServiceServer() *ServiceServer {
+	return &ServiceServer{}
+}
+
+func (s ServiceServer) CreateTask(ctx context.Context, req *api.CreateTaskRequest) (*api.CreateTaskResponse, error) {
+	log := zlog.Ctx(ctx)
+	db := hwgorm.GetDB(ctx)
+
+	// TODO: Auth
+
+	patientId, err := uuid.Parse(req.PatientId)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	// Check if patient exists
+	if err := db.First(intPatient.Patient{Id: patientId}).Error; err != nil {
+		return nil, status.Error(codes.NotFound, err.Error())
```

Comment:
the error here may be arbitrary (i.e. connectivity issues), but we always return a NotFound. This will make our lives harder when monitoring. (Happens a bunch of times here)

Reply:
Yes, right. I've adopted the `hwgorm.IsOurFault` helper. Thanks for the suggestion.

Reply:
We should commit to an error pattern and apply that across the services. Right now, it's not that consistent.